sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSensor.java

```diff
@@ -101,6 +101,7 @@ public void execute(SensorContext context) {
 
     sonarComponents.setSensorContext(context);
     sonarComponents.setCheckFilter(createCheckFilter(sonarComponents.isAutoScanCheckFiltering()));
+    sonarComponents.setConfiguration(settings);
```

Comment:
When we call `sonarComponents.setSensorContext(context);` we already provide `context.config()` so we don't need `sonarComponents.setConfiguration(settings);`.
Why do we need `setConfiguration`?